tests/unit/amo/utils/test_index.js

```diff
@@ -67,7 +67,19 @@ describe(__filename, () => {
     });
 
     describe('isInternal prop', () => {
-      it('returns true for a URL containing the current host', () => {
+      it('returns true for a single slash-prefixed URL', () => {
+        expect(checkInternalURL({ urlString: pathname }).isInternal).toEqual(
+          true,
+        );
+      });
+
+      it('returns false for a double slash-prefixed URL', () => {
+        expect(
+          checkInternalURL({ urlString: '//path/name' }).isInternal,
+        ).toEqual(false);
+      });
+
```

Comment:
Do we have a test case for an internal protocol-free URL?

Reply:
We don't actually support internal protocol-free URLs with the current code. We're using `url.parse` to extract the `host` from the URL (to compare with `baseURL` from `config`), but `url.parse` does not support protocol-free URLs. It just interprets the entire URL as the `path`. I guess if we truly want to support protocol-free URLs we need to write some extra code to deal with this edge-case.

On the other hand, one could argue that we shouldn't ever *need* to support them because of the source of these URLs. They are either
a) URLs that we are generating ourselves in the app (like an add-on URL), and I don't see any reason why we'd ever generate a protocol-free URL, or
b) URLs that are entered into the admin for the hero shelves. If a protocol-free URL is entered into the admin, the API will return it to us with a protocol prepended.

Therefore, I don't think we need to support them in this function, so I will remove the code that I added specifically for them.

Reply:
I actually do think that we should support protocol free urls, it will be more future proof and that is almost done.

Reply:
> I actually do think that we should support protocol free urls, it will be more future proof and that is almost done anyway.

Do you also think we need to support internal protocol free urls? I can't see a use case for someone specifically creating a protocol free url that points back to the site? I guess we can do it, but it's extra code to cover a use case that seems like it will never happen.

Reply:
I had the feeling that protocol free internal urls were already supported. If it is, great. If not, then let's add a comment to mention this limitation.

Reply:
Well, now you've got me thinking that if we're going to support it at all, we should probably support it fully. It's a bit of extra code, but I'm going to have ti support both cases (internal and external).